scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1762,10 +1766,16 @@ class Router(formatOps: FormatOps) {
       case FormatToken(c: T.Comment, _, _) =>
         Seq(Split(getMod(formatToken), 0))
 
-      case FormatToken(_: T.KwImplicit, _, _)
+      case FormatToken(soft.ImplicitOrUsing(), _, _)
           if !style.binPack.unsafeDefnSite &&
-            !style.verticalMultiline.atDefnSite =>
-        opensImplicitParamList(prevNonComment(prev(formatToken))).fold {
+            !style.verticalMultiline.atDefnSite &&
+            isRightImplicitOrUsingSoftKw(formatToken, soft) =>
+        val argsOrParamsOpt = formatToken.meta.leftOwner match {
+          // for the using argument list
+          case _: ApplyUsing => Some(getApplyArgs(formatToken, false).args)
```

Comment:
here, you're checking for the owner of `using`. why can't the same be done in the newly added method you call just above?

Reply:
the reason is that `isRightImplicitOrUsingSoftKw` returns true both for `using` at the definition site and at the call site. this match/case separates these because `using` def site doesn't have its `ApplyUsing` counterpart.

Reply:
thank you, understood!